stmhal/usbd_conf.c

```diff
@@ -379,90 +383,97 @@ void HAL_PCD_DisconnectCallback(PCD_HandleTypeDef *hpcd)
 USBD_StatusTypeDef  USBD_LL_Init (USBD_HandleTypeDef *pdev)
 { 
 #if defined(USE_USB_FS)
+if (pdev->id == 0 )

```

Comment:
Is there another way to detect FS vs HS?  By comparing some pointer (eg ->pData) to USB_FS / USB_HS?